inc/classes/rest/class-patterns.php

```diff
@@ -0,0 +1,906 @@
+
+		return new WP_REST_Response( $child_sites, 200 );
+	}
+
+	/**
+	 * Push patterns to target sites.
+	 *
+	 * @param WP_REST_Request $request The REST request object.
+	 *
+	 * @return WP_Error|WP_REST_Response
+	 */
+	public function push_patterns_to_targets( WP_REST_Request $request ) {
+
+		if ( ! Utils::is_governing_site() ) {
+			return new WP_Error( 'not_parent_site', __( 'This site is not configured as a parent site.', 'onedesign' ), array( 'status' => 403 ) );
+		}
+
+		$pattern_names = $request->get_param( 'pattern_names' );
+		// Expecting numeric ids based on your settings UI.
+		$target_site_ids = $request->get_param( 'target_site_ids' );
+
+		// Use the option name from your settings class.
+		$configured_child_sites = $this->get_compatible_sites_object();
+
+		// Get all patterns (both registered and user-created).
+		$local_patterns_map = $this->get_all_local_patterns_map();
+
+		$patterns_to_push = array();
+		foreach ( $pattern_names as $name ) {
+			if ( isset( $local_patterns_map[ $name ] ) ) {
+				$pattern_data       = $local_patterns_map[ $name ];
+				$patterns_to_push[] = array(
+					'name'          => $pattern_data['name'] ?? '',
+					'title'         => $pattern_data['title'] ?? '',
+					'content'       => $pattern_data['content'] ?? '',
+					'description'   => $pattern_data['description'] ?? '',
+					'categories'    => $pattern_data['categories'] ?? array(),
+					'keywords'      => $pattern_data['keywords'] ?? array(),
+					'viewportWidth' => $pattern_data['viewportWidth'] ?? null,
+					'blockTypes'    => $pattern_data['blockTypes'] ?? null,
+					'postTypes'     => $pattern_data['postTypes'] ?? null,
+					'templateTypes' => $pattern_data['templateTypes'] ?? null,
+					'inserter'      => $pattern_data['inserter'] ?? true,
+					'source'        => $pattern_data['source'] ?? 'registered',
+					'id'            => $pattern_data['id'] ?? null, // For user patterns.
+				);
+			}
+		}
+
+		if ( empty( $patterns_to_push ) ) {
+			return new WP_Error( 'no_patterns_selected', __( 'No valid patterns selected to push.', 'onedesign' ), array( 'status' => 400 ) );
+		}
+
+		$results           = array();
+		$current_site_name = get_bloginfo( 'name' );
+
+		foreach ( $target_site_ids as $site_id ) {
+			$target_site = array_filter(
+				$configured_child_sites,
+				function ( $site ) use ( $site_id ) {
+					return isset( $site['id'] ) && $site['id'] === $site_id;
+				}
+			);
+
+			$target_site = reset( $target_site ); // Get the first match, should be unique by ID.
+
+			if ( empty( $target_site ) ) {
+				$results[ $site_id ] = array(
+					'success' => false,
+					'message' => __( 'Target site not found in configuration.', 'onedesign' ),
+				);
+				continue;
+			}
+
+			// The 'api_key' in $target_site is the token for the remote child site.
+			$remote_api_key = $target_site['api_key'] ?? '';
+			$remote_url     = trailingslashit( $target_site['url'] ) . 'wp-json/onedesign/v1/receive-patterns';
```

Comment:
Let's define `trailingslashit( $target_site['url'] ) . 'wp-json/onedesign/v1/` as a class property variable in this class and use it all the places in this class.